### PR TITLE
Fix 'simplify conditional expression' with VB value equality

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/57472

Should be reviewed as two separate commits.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicSyntaxGeneratorInternal.vb [hunk 1]
```diff
@@ -126,6 +127,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
...
+            ElseIf negatedKind = BinaryOperatorKind.NotEquals Then
+                If node.IsKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression) Then
+                    Return generator.ValueNotEqualsExpression(left, right)
+                Else
+                    Return generator.ReferenceNotEqualsExpression(left, right)
+                End If
+            Else
+                Throw ExceptionUtilities.UnexpectedValue(negatedKind)
+            End If
+        End Function
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicSyntaxGeneratorInternal.vb:146]: this is the main part of the fix.  In VB, when negating an equality expression, we defer to the original syntax to decide what to negate to.  so `=` becomes `<>` and `Is` becomes `IsNot` (and vice versa).

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs [hunk 2]
```diff
@@ -130,48 +127,44 @@ internal static partial class SyntaxGeneratorExtensions
             }
 
             if (!s_negatedBinaryMap.TryGetValue(binaryOperation.OperatorKind, out var negatedKind))
-            {
                 return generator.LogicalNotExpression(expressionNode);
-            }
-            else
+
+            var negateOperands = false;
```
Comment (davidwengier) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs:132]: Why did this one not get the switch expression treatment? Or even just assign it to a pattern?

**Review by davidwengier — APPROVED** [review]
> Should be reviewed as two separate commits.

image

🤷‍♂️

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs:132]: I'll change :-)

**Review by ryzngard — APPROVED** [review]

**Pull request merged**